compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -545,6 +546,11 @@ impl<'a> Parser<'a> {
         self.check_or_expected(self.token.can_begin_const_arg(), TokenType::Const)
     }
 
+    fn check_inline_const(&mut self) -> bool {
+        self.check_keyword(kw::Const)
+            && self.look_ahead(1, |t| t == &token::OpenDelim(DelimToken::Brace))
```

Comment:
`token::NtBlock` should probably also work as a next token.
(At least it will be successfully parsed by `parse_const_expr`.)